projects/fsastorefrontlib/src/cms-components/assets-table/resolve-asset-value.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { TranslatePipe } from '@spartacus/core';
+import { RequestType } from '../../occ';
+
+@Pipe({ name: 'resolveAssetValue' })
+export class ResolveAssetValuePipe implements PipeTransform {
+  constructor(private tpipe: TranslatePipe) {}
+
+  transform(value: any, ...args: any[]) {
+    const asset = args[0];
+
+    if (value.propName) {
+      return value.value
+        .split('.')
+        .reduce((accum, curr) => accum?.[curr], asset);
+    }
+
+    const allowedFSRequestTypes = asset.categoryData.allowedFSRequestTypes;
+
+    return allowedFSRequestTypes
```

Comment:
Perhaps move this logic to separate method or variable and then return it's result? It would be easier to read if nothing else.